test/integration/render-tests/image-fallback-nested/feature-state/add-image/style.json

```diff
@@ -0,0 +1,137 @@
+{
+  "version": 8,
+  "metadata": {
+    "test": {
+      "width": 64,
+      "height": 64,
+      "allowed": 0.0015,
+      "operations": [
+        [
+          "addImage",
+          "rocket",
+          "./image/rocket.png"
+        ],
+                [
+          "wait"
+        ],
+        [
+          "addLayer",
```

Comment:
addLayer will trigger tile re-layout. Would it be possible to move layer definition to style definition? List of operations may be:

1. wait
2. add image
3. set feature state

Reply:
Thank you @alexshalamov!

I tried the test with the layer definition in the style definition as you describe (in place of the current  `"layers": []`. This causes the test to pass in GL Native but fail in GL JS. I think this is from an issue unique to render tests in GL JS.

I see two ways around this:
- resolve this issue in GL JS to get consistent render test behavior
- add the test in GL Native only (and perhaps open an issue to address this in GL JS at a later date).

I think the latter in the simplest approach for now, what do you think?

Reply:
I just tested out the changes you suggest, and they still don't seem to use the intended [updateVertexVectors](https://github.com/mapbox/mapbox-gl-native-internal/blob/3400ba5ecc829838e88f6d3d200f0076be7d21fb/src/mbgl/renderer/paint_property_binder.hpp#L799) function in Native.